modin/pandas/base.py

```diff
@@ -2322,31 +2323,31 @@ def check_dtype(t):
                             )
                         )
         else:
-            # Normally pandas returns this near the end of the quantile, but we
-            # can't afford the overhead of running the entire operation before
-            # we error.
-            if not any(is_numeric_dtype(t) for t in self._get_dtypes()):
-                raise ValueError("need at least one array to concatenate")
+            numeric_only_df = self.drop(
+                columns=[
+                    i for i in self.dtypes.index if not is_numeric_dtype(self.dtypes[i])
+                ]
+            )
```

Comment:
This is the second problem, due to the fact that we did not drop non numeric columns, internal and external index did not match.

`numeric_only=True` case has not been tested before.